lib/src/solver/solve_suggestions.dart

```diff
@@ -0,0 +1,214 @@
+    final bestRelease =
+        await inferBestFlutterRelease({cause.sdk.identifier: constraint});
+    if (bestRelease == null) return null;
+    final result = await tryResolve(type, cache, entrypoint.root,
+        lockFile: entrypoint.lockFile,
+        unlock: unlock,
+        sdkOverrides: {
+          'dart': bestRelease.dartVersion,
+          'flutter': bestRelease.flutterVersion
+        });
+    if (result != null) {
+      return _ResolutionSuggestion(
+        runningFromFlutter
+            ? '* Try using the Flutter SDK version: ${bestRelease.flutterVersion}. '
+            :
+            // Here we assume that any Dart version included in a Flutter
+            // release can also be found as a released Dart SDK.
+            '* Try using the Dart SDK version: ${bestRelease.dartVersion}. See https://dart.dev/get-dart.',
+      );
+    }
+    return null;
+  }
+
+  Future<_ResolutionSuggestion?> suggestSinglePackageUpdate(String name) async {
+    final originalConstraint = (entrypoint.root.dependencies[name] ??
+            entrypoint.root.devDependencies[name])
+        ?.constraint;
+    if (originalConstraint != null) {
+      final relaxedPubspec = stripVersionBounds(entrypoint.root.pubspec,
+          stripOnly: [name], stripLowerBound: true);
+
+      final result = await tryResolve(
+          type, cache, Package.inMemory(relaxedPubspec),
+          lockFile: entrypoint.lockFile, unlock: unlock);
+      if (result != null) {
+        final resolvingPackage =
+            result.packages.firstWhere((p) => p.name == name);
+
+        final addDescription =
+            packageAddDescription(entrypoint, resolvingPackage);
+
+        var priority = 1;
+        var suggestion =
+            '* Try updating your constraint on $name: $topLevelProgram pub add $addDescription';
+        if (originalConstraint is VersionRange) {
+          final min = originalConstraint.min;
+          if (min != null) {
+            if (resolvingPackage.version < min) {
+              priority = 3;
+              suggestion =
+                  '* Consider downgrading your constraint on $name: $topLevelProgram pub add $addDescription';
+            } else {
+              priority = 2;
+              suggestion =
+                  '* Try upgrading your constraint on $name: $topLevelProgram pub add $addDescription';
+            }
+          }
+        }
+
+        return _ResolutionSuggestion(suggestion, priority: priority);
+      }
+    }
+    return null;
+  }
+
+  Future<_ResolutionSuggestion?> suggestUnlockingAll(
+      {required bool stripLowerBound}) async {
+    final originalPubspec = entrypoint.root.pubspec;
+    final relaxedPubspec =
+        stripVersionBounds(originalPubspec, stripLowerBound: stripLowerBound);
+
+    final result = await tryResolve(
+        type, cache, Package.inMemory(relaxedPubspec),
+        lockFile: entrypoint.lockFile, unlock: unlock);
+    if (result != null) {
```

Comment:
```suggestion
      if (result == null) {
        return;
      }
```
Then we get less indentation...

Reply:
Done